atoti-query-analyser/src/components/Summary/Summary.tsx

```diff
@@ -200,6 +201,96 @@ function MapView<K, V>({
   );
 }
 
+type ComponentTimings = Readonly<{
+  database: number;
+  engine: number;
+  network: number;
+  providers: number;
+}>;
+
+const computeComponentTimings = (graph: RetrievalGraph): ComponentTimings => {
+  const operations = Array.from(graph.getVertices()).map((operation) =>
+    operation.getMetadata()
+  );
+  const databaseTime = operations
+    .filter((operation) => operation.type.includes("JIT"))
+    .flatMap((operation) => operation.timingInfo.elapsedTime ?? [])
+    .reduce((result, value) => result + value, 0);
+  const networkTime = operations
+    .flatMap((operation) => operation.timingInfo.broadcastingTime ?? [])
+    .reduce((result, value) => result + value, 0);
+  const providerTime = operations
+    .filter((operation) => operation.type.includes("Partial"))
+    .flatMap((operation) => operation.timingInfo.elapsedTime ?? [])
+    .reduce((result, value) => result + value, 0);
+  const engineTime = operations
+    .filter(
+      ({ type }) =>
+        !type.includes("Partial") &&
+        !type.includes("JIT")
+    )
+    .flatMap((operation) => operation.timingInfo.elapsedTime ?? [])
+    .reduce((result, value) => result + value, 0);
+  return {
+    network: networkTime,
+    database: databaseTime,
+    providers: providerTime,
+    engine: engineTime,
+  };
+};
+
+const createTimeFormatter = (
```

Comment:
Do you think that usage of the same formatter for all values is better than treating each value individually?

Reply:
After a live discussion with @ogorodnikoff2012 , we concluded that it was a matter of taste. Some may be interested in knowing the exact time, while a rounding is also ok to show elements in the same order.
We will start this way and switch to something more fancy if needed.